bootloader/bootloader.go

```diff
@@ -371,17 +380,22 @@ func extractKernelAssetsToBootDir(dstDir string, snapf snap.Container, assets []
 			return err
 		}
 	}
-	return nil
-}
 
-func removeKernelAssetsFromBootDir(bootDir string, s snap.PlaceInfo) error {
-	// remove the kernel blob
-	blobName := s.Filename()
-	dstDir := filepath.Join(bootDir, blobName)
-	if err := os.RemoveAll(dstDir); err != nil {
+	// and move into place
+	if err := os.Rename(dstDir, assetsDir); err != nil {
```

Comment:
we have `osutil.AtomicRename()` which does effectively the same thing, but in a more safe manner, can we use that instead here? That also deals with syncing directories as needed

Reply:
do we know how well these things work on vfat?

Reply:
hmm, actually I think we neglected to use AtomicRename for the extracted run kernel image bootloader stuff for some reason, but I don't think it was related to vfat problems

Reply:
Happy to change it to use that for clarity. Fwiw (and I know I'm super pedantic here) I don't think using it is safer, it's exactly the same code as used here (os.Rename+dir.Sync) unless I miss something.

Reply:
I mean personally if we have the code already written and tested in the helper it makes sense to me that we use the helper - if we don't have the desire to use the helper even in cases like this then we should probably just delete the helper